launch.m

```diff
@@ -247,90 +268,167 @@
+								'npy_matlab_repository', npy_matlab_repository,...
+								'catGTPath', catGTPath,...
+								'tPrime_path', tPrime_path,...
+								'cWaves_path', cWaves_path,...
+								'kilosort_output_tmp', kilosort_output_tmp,...
+								'npx_directory', npx_directory,...
+                'continuous_file', continuous_file,...
+                'spikeGLX_data', true,...
+                'input_meta_path', input_meta_fullpath,...
+                'kilosort_output_directory', kilosort_output_dir,...
+                'noise_template_use_rf', false,...
+								'catGT_run_name', session_ids{i},...
+                'gate_string', spec{2},...
+                'probe_string', prb,...
+            		'ks_remDup', ks_remDup,...                   
+                'ks_finalSplits', 1,...
+                'ks_labelGood', 1,...
+							  'ks_saveRez', ks_saveRez,...
+							  'ks_copy_fproc', ks_copy_fproc,...
+							  'ks_minfr_goodchannels', ks_minfr_goodchannels,...                  
+							  'ks_whiteningRadius_um', ks_whiteningRadius_um,...
+							  'ks_Th', ks_Th,...
+							  'ks_CSBseed', 1,...
+							  'ks_LTseed', 1,...
+							  'ks_templateRadius_um', ks_templateRadius_um,...
+							  'extracted_data_directory', catGT_dest,...
+							  'event_ex_param_str',  event_ex_param_str,...
+							  'c_Waves_snr_um', c_Waves_snr_um,...                   
+                'qm_isi_thresh', (refPerMS/1000)...
+						)...
+        );
+
+				% copy json file to data directory as record of the input parameters 
     end
     
+		% loop over probes for processing.
+		for i=1:length(prob_list)
+				prb = prob_list{i};
+
+        py.py_modules.caller.runOne(...
+						session_ids{i},...
+            json_directory,...
+            data_directory{i},...
+            run_CatGT,...
+            catGT_input_json{i},...
+            catGT_output_json{i},...
+            modules,...
+            module_input_json{i},...
+            logFullPath...
+				);
+		end
 end
 
-% generate json config file for each job
-%py.help('py_modules.caller.createInputJson');
-%py.py_modules.caller.createInputJson(...
-%    pyargs(...
-%        'output_file', 'configs/test.json',...
-%        'npx_directory', npx_directory,...
-%        'continuous_file', 'continuous_file path',...
-%        'kilosort_output_directory', catGT_dest,...
-%        'spikeGLX_data', true,...
-%        'input_meta_path', 'input_meta_fullpath',...
-%        'catGT_run_name', 'spec(0)',...
-%        'gate_string', 'spec(1)',...
-%        'trigger_string', 'trigger_str',...
-%        'probe_string', 'prb',...
-%        'catGT_stream_string', 'catGT_stream_string',...
-%        'catGT_car_mode', car_mode,...
-%        'catGT_loccar_min_um', loccar_min,...
-%        'catGT_loccar_max_um', loccar_max,...
-%        'catGT_cmd_string', strcat('catGT_cmd_string', ' ', 'extract_string'),...
-%        'extracted_data_directory', catGT_dest...
-%    )...
-%);
-
 
 %% Pipeline Configuration
```

Comment:
If launch.m is a sort of "example" script maybe it would be good to be more verbose for user to understand class structure. For example, `npxtoolkit.jobs.KiloSort()` etc... instead of `KiloSort()`.

Reply:
Most of the launch.m will be moved to individual job classes. We can do a cleanup code review on launch.m by that time.